checks/checks.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2023 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checks
+
+import "github.com/ossf/scorecard/v4/checker"
+
+type Check interface {
+	Run(c *checker.CheckRequest) checker.CheckResult
+}
```

Comment:
Some other questions that popped up regarding things associated with checks:
* Should supported request types be part of the interface?
* What about `Name`?
* Risk?

Reply:
I am maintaining the current check function design. We should review it when we make alterations.

Reply:
👍